### PR TITLE
refactor: avoid validation for table constraints during parse

### PR DESCRIPTION
**Pull request by ShiKaiWi**
# Which issue does this PR close?

Closes #

# Rationale for this change
Disallow for mutiple definitions of timestamp key is checked and should be checked by `planner`: https://github.com/CeresDB/ceresdb/blob/75e9b7ea549cde7cd906f008a3879ebed6a9db57/sql/src/planner.rs#L294

# What changes are included in this PR?
- Avoid validation for exact only one timestamp key definition during parsing procedure.
- Add test for defining multiple timestamp key constraints.

# Are there any user-facing changes?
No.

# How does this change test
Integration test.

### CODE REVIEW

**Comment by ShiKaiWi** [comment]
@dust1 I find that the disallow for multiple timestamp keys definition has been checked in planner module so there is no need to do that in the parser module. Sorry for misleading you and PTAL at this PR.

**Comment by dust1** [comment]
that's okay.😃
do I need to submit a new PR to remove the verification in the parser module? Or is it up to you to remove it in this PR?

File: tests/cases/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+*.out
```
Comment (waynexia) [inline tests/cases/.gitignore:1]: don't ignore the `.out` file. It should be tracked and we need to guarantee no `.out` will be added.

File: sql/src/parser.rs [hunk 2]
```diff
@@ -504,63 +501,30 @@ impl<'a> Parser<'a> {
...
-    }
-    Err(ParserError::ParserError(
-        "Failed to parser, not allowed define mutiple tskey constraints.".to_string(),
-    ))
-}
-
-fn try_build_tskey_constraint(col_defs: &[ColumnDef]) -> Result<Option<TableConstraint>> {
-    let mut constraint = None;
+// Build the tskey constraint from the column definitions if any.
+fn try_build_timestamp_key_constraint(
```
Comment (waynexia) [inline sql/src/parser.rs:505]: the `try_` prefix in fn name is not appropriate since this function won't fail

**Review by waynexia — COMMENTED** [review]
👍

**Comment by ShiKaiWi** [comment]
> that's okay.😃 do I need to submit a new PR to remove the verification in the parser module? Or is it up to you to remove it in this PR?

Let this pr remove it. Thanks.

File: tests/cases/.gitignore (hunk 1 above)
Comment (ShiKaiWi) [reply tests/cases/.gitignore:1]: Should it be guaranteed that no `.out` is added if we ignore it?

Comment (ShiKaiWi) [reply tests/cases/.gitignore]: I see. We can find the test failure by checking the existence of these `.out` files.

**Review by waynexia — APPROVED** [review]
Good job 👍

**Pull request merged**